File: sdks/dotnet/src/Resources/CustomersResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class CustomersResource
    {
        private readonly PuxbayClient _client;

        public CustomersResource(PuxbayClient client)
        {
            _client = client;
        }

        public async Task<PaginatedResponse<Customer>> ListAsync(int page = 1, int pageSize = 20)
        {
            return await _client.GetAsync<PaginatedResponse<Customer>>($"customers/?page={page}&page_size={pageSize}");
        }

        public async Task<Customer> GetAsync(string customerId)
        {
            return await _client.GetAsync<Customer>($"customers/{customerId}/");
        }

        public async Task<Customer> CreateAsync(Customer customer)
        {
            return await _client.PostAsync<Customer>("customers/", customer);
        }

        public async Task<Customer> UpdateAsync(string customerId, Customer customer)
        {
            return await _client.PatchAsync<Customer>($"customers/{customerId}/", customer);
        }

        public async Task DeleteAsync(string customerId)
        {
            await _client.DeleteAsync($"customers/{customerId}/");
        }
    }
}

File: sdks/php/src/Resources/CustomersResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Customer;
use Puxbay\Models\PaginatedResponse;

class CustomersResource extends BaseResource
{
    public function list(int $page = 1, int $pageSize = 20): PaginatedResponse
    {
        $data = $this->get("customers/?page={$page}&page_size={$pageSize}");
        return PaginatedResponse::fromArray($data, fn($item) => Customer::fromArray($item));
    }

    public function getCustomer(string $customerId): Customer
    {
        $data = $this->get("customers/{$customerId}/");
        return Customer::fromArray($data);
    }

    public function create(Customer $customer): Customer
    {
        $data = $this->post('customers/', $customer->toArray());
        return Customer::fromArray($data);
    }

    public function update(string $customerId, Customer $customer): Customer
    {
        $data = $this->patch("customers/{$customerId}/", $customer->toArray());
        return Customer::fromArray($data);
    }

    public function deleteCustomer(string $customerId): void
    {
        $this->delete("customers/{$customerId}/");
    }

    public function adjustLoyaltyPoints(string $customerId, int $points, string $description): Customer
    {
        $data = $this->post("customers/{$customerId}/adjust-loyalty-points/", [
            'points' => $points,
            'description' => $description,
        ]);
        return Customer::fromArray($data);
    }
}
